### ISSUE TITLE
hangs with `^` regex

### ISSUE THREAD

**Issue by ponyatov** [loloicci/nimly#51]
```nim
variant MyToken:
  tSYM(str: string)

niml lexer[MyToken]:
  r"\S+": tSYM(token.token)
```
```
nimble --cc:tcc build
  Verifying dependencies for kiosk@0.1.0
      Info: Dependency on nimly@any version already satisfied
  Verifying dependencies for nimly@0.6.0
      Info: Dependency on patty@>= 0.3.3 already satisfied
  Verifying dependencies for patty@0.3.3
   Building kiosk/kiosk using c backend
...
```
Long delay about 10-15 seconds, then error follows:
```
        ... Command: "/home/ponyatov/.nimble/bin/nim" c --noNimblePath -d:NimblePkgVersion=0.1.0 --cc:tcc --path:"/home/ponyatov/.nimble/pkgs/nimly-0.6.0"  --path:"/home/ponyatov/.nimble/pkgs/patty-0.3.3"  -o:"/home/ponyatov/metaL/kiosk/kiosk" "/home/ponyatov/metaL/kiosk/src/kiosk.nim"
        ... stack trace: (most recent call last)
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(33, 11)
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(33, 11) :tmp
        ... /home/ponyatov/.nimble/pkgs/nimly-0.6.0/nimly/lexgen.nim(595, 43) convertToLexData
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/pure/collections/tables.nim(846, 15) []
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/pure/collections/tables.nim(262, 7) []
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(30, 6) template/generic instantiation of `niml` from here
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/pure/collections/tables.nim(262, 7) Error: unhandled exception: key not found: -1 [KeyError]
```

**Comment by ponyatov** [comment]
With other variant
```nim
niml lexer[MyToken]:
  r"[^ \t\r\n]+": tSYM(token.token)
```
no delay before error
```
        ... stack trace: (most recent call last)
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(34, 19)
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(34, 19) :tmp
        ... /home/ponyatov/.nimble/pkgs/nimly-0.6.0/nimly/lexgen.nim(895, 26) convertToReSynTree
        ... /home/ponyatov/.nimble/pkgs/nimly-0.6.0/nimly/lexgen.nim(705, 11) convertToSeqRePart
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/system/assertions.nim(29, 26) failedAssertImpl
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/system/assertions.nim(22, 11) raiseAssert
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/system/fatal.nim(49, 5) sysFatal
        ... /home/ponyatov/metaL/kiosk/src/kiosk.nim(30, 6) template/generic instantiation of `niml` from here
        ... /home/ponyatov/.choosenim/toolchains/nim-1.2.4/lib/system/fatal.nim(49, 5) Error: unhandled exception: /home/ponyatov/.nimble/pkgs/nimly-0.6.0/nimly/lexgen.nim(705, 18) `false` invalid escaping in class [AssertionError]
```

**Comment by ponyatov** [comment]
```
ponyatov@meteo:~/metaL/kiosk$ nim -version
Nim Compiler Version 1.2.4 [Linux: amd64]
Compiled at 2020-06-26
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release
```

**Comment by loloicci** [comment]
@ponyatov Thank you for reporting!
This bug had fixed in v0.6.1 with #52 .
